test/incrParse/inserted_text_starts_identifier.swift

```diff
@@ -0,0 +1,6 @@
+// RUN: %empty-directory(%t)
+// RUN: %validate-incrparse %s --test-case STRING
+
+// SR-8995 rdar://problem/45259469
+
+self = <<STRING<|||_                            _>>>foo(1)[object1, object2] + o bar(1)
```

Comment:
In this test case, the second `_` inserted used to be translated to the "pre-edit" position of `bar()`. So the parser tried to reuse `bar()` when parsing `_foo(1)...`